src/csvcubed/utils/sparql_handler/data_cube_state.py

```diff
@@ -1,57 +1,50 @@
+"""
+Data Cube State
+---------------
+
+Provides access to inspect the contents of an rdflib.ConjunctiveGraph containing
+one of more data cubes.
+"""
+
 from dataclasses import dataclass
 from functools import cached_property
+from pathlib import Path
 from typing import Dict, List, Optional, TypeVar
+from urllib.parse import urljoin
 
 from csvcubed.models.cube.cube_shape import CubeShape
 from csvcubed.models.sparqlresults import (
-    ColumnDefinition,
     CubeTableIdentifiers,
     IsPivotedShapeMeasureResult,
     QubeComponentsResult,
     UnitResult,
 )
+from csvcubed.utils.dict import get_from_dict_ensure_exists
 from csvcubed.utils.iterables import first, group_by
 from csvcubed.utils.sparql_handler.csvw_state import CsvWState
+from csvcubed.utils.sparql_handler.sparql import path_to_file_uri_for_rdflib
 from csvcubed.utils.sparql_handler.sparqlquerymanager import (
-    select_column_definitions,
     select_csvw_dsd_qube_components,
     select_data_set_dsd_and_csv_url,
     select_is_pivoted_shape_for_measures_in_data_set,
     select_units,
 )
+from csvcubed.utils.uri import looks_like_uri
 
 T = TypeVar("T")
 
 
 @dataclass
 class DataCubeState:
-    csvw_state: CsvWState
-
-    """
-    Private utility functions.
-    """
-
-    def _get_value_for_key(self, key: str, dict: Dict[str, T]) -> T:
-        maybe_value = dict.get(key)
-        if maybe_value is None:
-            raise KeyError(f"Could not find the definition for key '{key}'")
-        return maybe_value
+    """todo: Describe the class"""
 
-    """
-    Private cached properties.
-    """
+    csvw_state: CsvWState
 
-    @cached_property
-    def _column_definitions(self) -> Dict[str, List[ColumnDefinition]]:
-        """
-        Map of csv_url to the list of column definitions for the given CSV file.
-        """
-        results = select_column_definitions(self.csvw_state.rdf_graph)
-        return group_by(results, lambda r: r.csv_url)
+    # Private cached properties.
 
     @cached_property
     def _units(self) -> Dict[str, UnitResult]:
-        """ """
+        """todo: Fill me in"""
```

Comment:
Needs a comment

Reply:
Doesn't look like this has been completed to me.